### ISSUE TITLE
Android remote debug issue

### ISSUE THREAD

**Issue by RoboLampDev** [godotengine/godot#82390]
### Godot version

4.1.2.rc1

### System information

Godot v4.1.2.rc1 - Windows 10.0.19045 - Vulkan (Mobile) - integrated AMD Radeon(TM) Vega 8 Graphics () - AMD Ryzen 3 2200G with Radeon Vega Graphics (4 Threads)

### Issue description

its android remote debug issue
after remote debug loading finish i got an error

-editor/export/editor_export_platform.h:178 - Export: Building of Android project failed, check output for the error. Alternatively visit docs.godotengine.org for Android build documentation.
-  Cannot remove non-existent file or directory: C:/Users/ASD/AppData/Local/Godot/tmpexport.1695701345.apk

### Steps to reproduce

-editor/export/editor_export_platform.h:178 - Export: Building of Android project failed, check output for the error. Alternatively visit docs.godotengine.org for Android build documentation.
-  Cannot remove non-existent file or directory: C:/Users/ASD/AppData/Local/Godot/tmpexport.1695701345.apk

### Minimal reproduction project

![Screenshot_195](https://github.com/godotengine/godot/assets/146053254/9b13ccdf-b31b-4882-94ee-7fd823199ab9)
![Screenshot_194](https://github.com/godotengine/godot/assets/146053254/3d9b0a03-d74e-4066-b7e2-47a208ba7556)

**Comment by nontoxicguy** [comment]
Same for me, I have Godot 4.1.1. To export for android I've followed the instructions in the docs, using the way that allows you to only install the apk tools and not android studio. The only thing that I did differently from the docs is that I installed Open JDK 17 instead of 11 due to some version issues that I don't remember well but that forced me to download Open JDK 17. I don't see how this could cause the error but I don't have any other idea of what could be causing this.

**Comment by duarteroso** [comment]
I've been having this message in two case:

1. My project was not compiling
2. Something in the Android Build Template I've added was not compiling

Of course there might be other cases I haven't seen.
I would suggested the following:

1. Make sure your project builds and runs on the editor (you probably have done it already)
2. Open the Android Build Template with Android Studio and make sure it builds there as well.

Hope it helps!

**Comment by nontoxicguy** [comment]
I managed to fix my issue. In short: install OpenJDK JDK version not JRE.

**Comment by DarnelleF** [comment]
> version

are you using windows? cause i have OpenJDK 11 and same issue

**Comment by nontoxicguy** [comment]
> > version
> 
> are you using windows? cause i have OpenJDK 11 and same issue

Yes, pretty funny that I was searching how to switch to linux when I noticed that you quoted me.

Anyways, if you're talking about the "version" issue I talked about well higher versions of JDK support higher versions of the thing that forced me to install JDK 17. In any case this "version" issue only appears when making the debug.keystore and the command line tools. I currently have JDK 11 installed and it works.

If you're talking about the thing this issue was opened for then look at the OpenJDK download page. Half of the downloads are marked JDK and the other hald JRE if I remember well. The JRE are much more lightweight but in my case downloading a JRE instead of a JDK was the root cause of my problem.

I'm going to take this oppotunity to say if we should update the docs. In the docs the version of JDK to install is not specified that much, but has a large impact.

**Comment by DarnelleF** [comment]
i am using mac m1. ive tried everything and cant figure it out

**Comment by nontoxicguy** [comment]
> i am using mac m1. ive tried everything and cant figure it out also i noticed that i told godot to use my custom android_sdk folder different from where android studio installs it but the error message states that the file is looking for is not in my custom android_sdk but where adnroid studios installs android so im getting more clues..... been trying to work this out for a month

The file you provide in editor setting export/android/android_sdk_path is looking like this?
![cmdline-tools](https://github.com/godotengine/godot/assets/140744676/501e8473-9700-472b-ab00-5e2e31fff84b)
I generated this folder with JRE 17 and switched to JDK 11 after creating it, then it worked for me.

**Comment by DarnelleF** [comment]
thanks, i will give it a try !
